internal/component/database_observability/postgres/collector/query_samples.go

```diff
@@ -137,6 +142,26 @@ type SampleState struct {
 	LastSeenAt  time.Time
 	LastCpuTime string // last cpu_time observed under CPU condition
 	tracker     WaitEventTracker
+	// EndOverride is used to compute durations and timestamps when a query
+	// transitioned to idle or was only observed as idle.
+	EndOverride sql.NullTime
+}
+
+func (s *SampleState) updateCpuTimeIfActive(sample QuerySamplesInfo) {
```

Comment:
```suggestion
func (s *SampleState) updateCPUTimeIfActive(sample QuerySamplesInfo) {
```

Reply:
I think we could just call this `updateCPUTime`, the fact we use the state is kind of an implementation detail